tests/integration/awslambda/test_lambda.py

```diff
@@ -696,6 +698,120 @@ def test_url_config_lifecycle(self, lambda_client, create_lambda_function, snaps
 
+    @pytest.mark.aws_validated
+    def test_large_environment_variables_fails(
+        self, lambda_client, create_lambda_function, snapshot
+    ):
+        """Lambda functions with environment variables larger than 4 KiB should fail to create."""
+        snapshot.add_transformer(snapshot.transform.lambda_api())
+
+        # set up environment mapping with a total size of 4 KB
+        strlen = lambda s: len(s.encode("utf8"))
+        key = "LARGE_VAR"
+        key_bytes = strlen(key)
+        #  need to reserve bytes for json encoding ({, }, 2x" and :). This is 7
+        #  bytes, so reserving 6 makes the environment variables one byte to
+        #  big.
+        target_size = 4 * KB - 6
+        large_envvar_bytes = target_size - key_bytes
+        large_envvar = "x" * large_envvar_bytes
+
+        function_name = "large-envvar-lambda"
+
+        with pytest.raises(lambda_client.exceptions.InvalidParameterValueException) as ex:
+            create_lambda_function(
+                handler_file=TEST_LAMBDA_PYTHON_ECHO,
+                func_name=function_name,
+                runtime=LAMBDA_RUNTIME_PYTHON39,
+                envvars={
+                    key: large_envvar,
+                },
+            )
+
+        snapshot.match("failured_create_fn_result", ex.value.response)
+        with pytest.raises(ClientError) as exc:
+            lambda_client.get_function(FunctionName=function_name)
+
+        assert exc.match("ResourceNotFoundException")
+
+    @pytest.mark.aws_validated
+    def test_large_environment_fails_multiple_keys(
+        self, lambda_client, create_lambda_function, snapshot
+    ):
+        """Lambda functions with environment variables larger than 4 KiB should fail to create."""
+        snapshot.add_transformer(snapshot.transform.lambda_api())
+
+        # set up environment mapping with a total size of 4 KB
+        env = {"SMALL_VAR": "ok"}
+        strlen = lambda s: len(s.encode("utf8"))
+        envlen = lambda e: strlen(json.dumps(e, separators=(":", ",")))
+
+        key = "LARGE_VAR"
+        # this size makes the environment > 4K
+        target_size = 4064
+        large_envvar = "x" * target_size
+        env[key] = large_envvar
+        assert envlen(env) == 4097
+
+        function_name = "large-envvar-lambda"
+
+        with pytest.raises(lambda_client.exceptions.InvalidParameterValueException) as ex:
+            create_lambda_function(
+                handler_file=TEST_LAMBDA_PYTHON_ECHO,
+                func_name=function_name,
+                runtime=LAMBDA_RUNTIME_PYTHON39,
+                envvars=env,
+            )
+
+        snapshot.match("failured_create_fn_result_multi_key", ex.value.response)
+
+        with pytest.raises(ClientError) as exc:
+            lambda_client.get_function(FunctionName=function_name)
+
+        assert exc.match("ResourceNotFoundException")
+
+    @pytest.mark.aws_validated
+    @pytest.mark.skip_snapshot_verify(
```

Comment:
```suggestion
    @pytest.mark.skip_snapshot_verify(
        condition=is_old_provider,
```

Reply:
You had no way to know, but we try to limit the skip_snapshot_verifies to the old provider, so we have a strong validation for the new one!